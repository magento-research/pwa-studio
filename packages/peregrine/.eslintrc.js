const config = {
    parser: 'babel-eslint',
    extends: ['@magento'],
    rules: {
        "no-undef": "off"
    }
};

module.exports = config;
