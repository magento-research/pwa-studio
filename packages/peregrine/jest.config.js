module.exports = {
    displayName: 'Peregrine',
    setupFiles: ['<rootDir>/scripts/shim.js', '<rootDir>/scripts/fetch-mock.js']
};
