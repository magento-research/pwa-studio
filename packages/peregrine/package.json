{
  "name": "@magento/peregrine",
  "version": "3.0.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "build": " ",
    "build:ci": " ",
    "build:cjs": " ",
    "build:dev": " ",
    "clean": " ",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": " "
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@storybook/react": "~4.1.13",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
