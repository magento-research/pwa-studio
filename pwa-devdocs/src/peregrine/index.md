---
title: Peregrine
---

The Peregrine project is a collection of UI Components for Magento PWA projects.
Use, extend, or remix these components to create a unique Magento PWA storefront.

Visit the [`Peregrine`] repository to contribute to this project.

[`Peregrine`]: https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine
