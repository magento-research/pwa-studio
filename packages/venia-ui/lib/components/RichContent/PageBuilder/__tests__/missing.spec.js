import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import Missing from '../missing';

jest.mock('../../../../classify');

test('renders a Missing component', () => {
    const missingProps = {
        contentType: 'test-content-type'
    };
    const component = createTestInstance(<Missing {...missingProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});
