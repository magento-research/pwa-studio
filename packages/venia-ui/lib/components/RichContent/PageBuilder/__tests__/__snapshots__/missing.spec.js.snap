// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Missing component 1`] = `
<div
  className="missing"
>
  <strong>
    Error:
  </strong>
   No component for 
  <strong>
    test-content-type
  </strong>
   content type.
</div>
`;
